### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by calvinrp**
- add a `read_to_end()`
- `Headers` is now an alias of the type from `wasi` crate
- `Request` can now set the HTTP `Headers`

### CODE REVIEW

File: src/http/client.rs [hunk 1]
```diff
@@ -41,10 +42,12 @@ impl<'a> Client<'a> {
...
-            res,
-            self.reactor.clone(),
-        )?)
+        Response::try_from_incoming_response(res, self.reactor.clone())
+    }
+
+    /// HTTP GET request.
+    pub async fn get(&self, url: Url) -> Result<Response<IncomingBody>> {
+        self.send(Request::new(Method::Get, url)).await
     }
```
Comment (yoshuawuyts) [inline src/http/client.rs:51]: Can we remove this? I understand this comes in handy when testing, but if we add this for the `Get` method we will want to add it for more methods too - and that ends up adding additional API surface.

File: src/http/response.rs [hunk 2]
```diff
@@ -148,3 +127,25 @@ impl AsyncRead for IncomingBody {
...
+                Ok(mut bytes) => buf.append(&mut bytes),
+                Err(StreamError::LastOperationFailed(err)) => {
+                    return Err(std::io::Error::other(err.to_debug_string()));
+                }
+                Err(StreamError::Closed) => break,
+            };
+        }
+
+        Ok(buf)
+    }
```
Comment (yoshuawuyts) [inline src/http/response.rs:150]: Ohh yeah, this is neat! - Reading the entire body into a vec is a common pattern.

Fhis would probably be a good fit to generalize as part of `wstd::io::Read` trait as an async dual to [`std::io::Read::read_to_end`](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_end). Could you add this there instead? That way all other types can reuse this functionality, as long as they implement the `Read` trait.

**Review by yoshuawuyts — CHANGES_REQUESTED** [review]
Two minor changes - but overall this looks good. Thank you!

**Comment by calvinrp** [comment]
Revising and following up with multiple PRs.

**Pull request closed without merging**